README.md

```diff
@@ -1,49 +1,52 @@
 # Lotus Eth JSON-RPC integration tests
 
-This project holds a suite of integration tests for the Ethereum JSON-RPC API
-built in Lotus with the Filecoin EVM runtime.
+This project holds a suite of integration tests for the Ethereum JSON-RPC API built in [Lotus](https://github.com/filecoin-project/lotus) with the Filecoin EVM runtime.
 
-This repo is the test runner for all the project repo in the `extern/` folder.
+This repo runs some basic tests direclty against ethers.js and web3.js libraries, and also runs deployment and tests in the [fevm-hardhat-kit](https://github.com/filecoin-project/fevm-hardhat-kit) (deploy only), [openzeppelin-contracts](https://github.com/filecoin-project/openzeppelin-contracts) and [fevm-uniswap-v3-core](https://github.com/DigitalMOB2/fevm-uniswap-v3-core) repositories, each of which use hardhat to deploy and test smart contracts on the Filecoin EVM.
+
+A custom version of Lotus is built to run these tests, which is included in the [`node`](./node) directory. Each build pulls the current `master` branch of Lotus, so the tests are always run against the latest version of Lotus.
+
+See [`Makefile`](./Makefile) for more details on how the tests are set up and executed.
 
 ## Running tests
 
-### Build the local lotus
+### Build the local lotus-runner
 
-1. Go through the [Lotus installation](https://lotus.filecoin.io/lotus/install/prerequisites/) guide to install the dependencies
-2. Install node dependencies by running `npm install`
-3. cd `./node` and run `make node`
-4. Run the node in `./bin/node`
+1. Ensure all dependencies are installed as described for a [Lotus installation](https://lotus.filecoin.io/lotus/install/prerequisites/)
+2. Run `make build-lotus-runner`
 
-### Prepare the .env file
+### Install the dependencies
 
-1. echo -n DEPLOYER_PRIVATE_KEY=0x > .env
-2. openssl rand -hex 32 >> .env
-3. echo -n USER_1_PRIVATE_KEY=0x >> .env
-4. openssl rand -hex 32 >> .env
-5. Copy the .env file into ./extern/fevm-hardhat-kit/, ./extern/openzeppelin-contracts/, ./extern/fevm-uniswap-v3-core/
+Run `make install`
+
+This will install dependencies for each of:
+1. The kit to interact with the lotus-runner (`make install-kit`)
+2. The local ethers.js and web3.js library tests (`make install-libs`)
+3. fevm-hardhat-kit (`make install-fevm-hardhat`)
+4. openzeppelin-contracts (`make install-openzeppelin`)
+5. fevm-uniswap-v3-core (`make install-uniswap-v3-core`)
 
 ### Run each test project
 
-1. ethers.js and web3.js (root folder)
-   1. npm install
-   2. npx hardhat --network itest test
-2. ./extern/fevm-hardhat-kit
-   1. change the `DEPLOYER_PRIVATE_KEY` in .env to `PRIVATE_KEY`
-   2. yarn
-   3. npx hardhat --network itest deploy
-3. ./extern/openzeppelin-contracts 
-   1. npm install
-   2. npx hardhat --network itest test
-4. ./extern/fevm-uniswap-v3-core
-   1. yarn
-   2. npx hardhat --network itest test
+1. Run the local lotus-runner: `make start-lotus-runner`
+2. *(In another terminal)* Run `make test`
+
```

Comment:
When I get here I run into an issue - my folder extern/fevm-hardhat-kit is empty. What should I do to build it properly?

```
mikers@mikers-B560-DS3H-AC-Y1:~/dev/rvagg/fevm-contract-tests$ make test
grep: extern/fevm-uniswap-v3-core/node_modules/@nomiclabs/hardhat-ethers/internal/provider-proxy.js: No such file or directory
sed: can't read extern/fevm-uniswap-v3-core/node_modules/@nomiclabs/hardhat-ethers/internal/provider-proxy.js: No such file or directory
Hardhat / ethers polling interval set to 200ms in extern/fevm-uniswap-v3-core/node_modules/@nomiclabs/hardhat-ethers/internal/provider-proxy.js
grep: ./extern/fevm-hardhat-kit/node_modules/@nomiclabs/hardhat-ethers/internal/provider-proxy.js: No such file or directory
sed: can't read ./extern/fevm-hardhat-kit/node_modules/@nomiclabs/hardhat-ethers/internal/provider-proxy.js: No such file or directory
Hardhat / ethers polling interval set to 200ms in ./extern/fevm-hardhat-kit/node_modules/@nomiclabs/hardhat-ethers/internal/provider-proxy.js
(cd libs-tests && \
npm exec -c "hardhat clean" && \
npm exec -c "hardhat --network itest test")
Resetting Lotus node with block time: 200
Setting up new wallet for: 0x6be45e32a1e57b195be925570d5c68d8f29dc6e08859400b34e0e8b4abcc06b7
Sent 1000 to 0x7c5d5Aa6c0e37EBEc9b69594089869496dECcd71
Finished setup, Node RPC endpoint: http://127.0.0.1:39215
Compiled 7 Solidity files successfully (evm target: paris).


  ethers.js ERC20
    ✔ Should successfully deploy (709ms)
    ✔ Should access deployment transaction receipt
    ✔ Should set the right name
    ✔ Should set the right symbol
    ✔ Should set the right initial supply
    ✔ Should be able to transfer token (624ms)

  ethers.js SimpleCoin
    ✔ Should send deployment transaction
    ✔ Should access transaction details before it has been mined
    ✔ Should access null transaction receipt before it has been mined
    ✔ Should successfully deploy (611ms)
    ✔ Should access transaction details after it has been mined
    ✔ Should access transaction receipt after it has been mined
    ✔ Should find the transaction in block tx list
    ✔ Should get block tx count
    ✔ Should interact with the contract using eth_call
    ✔ Should get the contract byte code at the deployed address
    - Should get storage using eth_getStorageAt

  web3.js ERC20
    ✔ Should successfully deploy (1110ms)
    ✔ Should set the right name
    ✔ Should set the right symbol
    ✔ Should set the right initial supply
    ✔ Should be able to transfer token (1051ms)

  web3.js SimpleCoin
    ✔ Should send deployment transaction (44ms)
    - Should access transaction details before it has been mined
    ✔ Should access null transaction receipt before it has been mined
    ✔ Should successfully deploy (1031ms)
    ✔ Should access transaction details after it has been mined
    ✔ Should access transaction receipt after it has been mined
    ✔ Should find the transaction in block tx list
    ✔ Should get block tx count
    ✔ Should interact with the contract using eth_call
    ✔ Should get the contract byte code at the deployed address
    - Should get storage using eth_getStorageAt


  30 passing (5s)
  3 pending

················································································································
|  Solidity and Network Configuration                                                                          │
···························|················|················|·················|································
|  Solidity: 0.8.23        ·  Optim: true   ·  Runs: 1000    ·  viaIR: false   ·   Block: 10,000,000,000 gas   │
···························|················|················|·················|································
|  Methods                                                                                                     │
···························|················|················|·················|················|···············
|  Contracts / Methods     ·  Min           ·  Max           ·  Avg            ·  # calls       ·  usd (avg)   │
···························|················|················|·················|················|···············
|  ERC20PresetFixedSupply  ·                                                                                   │
···························|················|················|·················|················|···············
|      transfer            ·     5,339,368  ·     5,503,031  ·      5,421,200  ·             2  ·           -  │
···························|················|················|·················|················|···············
|  Deployments                              ·                                  ·  % of limit    ·              │
···························|················|················|·················|················|···············
|  ERC20PresetFixedSupply  ·    30,597,099  ·    31,062,440  ·     30,752,213  ·         0.3 %  ·           -  │
···························|················|················|·················|················|···············
|  SimpleCoin              ·    14,062,174  ·    14,530,369  ·     14,296,272  ·         0.1 %  ·           -  │
···························|················|················|·················|················|···············
|  Key                                                                                                         │
················································································································
|  ◯  Execution gas for this method does not include intrinsic gas overhead                                    │
················································································································
|  △  Cost was non-zero but below the precision setting for the currency display (see options)                 │
················································································································
|  Toolchain:  hardhat                                                                                         │
················································································································
grep: extern/fevm-hardhat-kit/hardhat.config.js: No such file or directory
sed: can't read extern/fevm-hardhat-kit/hardhat.config.js: No such file or directory
Inserted itest network configuration in extern/fevm-hardhat-kit/hardhat.config.js
(cd extern/fevm-hardhat-kit && \
rm -rf deployments/ && \
npm exec -c "hardhat clean" && \
npm exec -c "hardhat --network itest deploy")
Error HH1: You are not inside a Hardhat project.

For more info go to https://hardhat.org/HH1 or run Hardhat with --show-stack-traces
make: *** [Makefile:85: test-fevm-hardhat] Error 1

```

Reply:
ah, the "fix-hardhat" target assumes you've installed everything first, but in theory you should be able to run just one of these

so, I've made it more relaxed about missing files, and I've also tied together the install targets with the test targets so you can't even not install before test so hopefully I'm using the potential of make more fully here and haven't tangled it up too much (as is always possible with make!)

Reply:
Thanks! I think my issue was running `make install` which ran `make .env` before checking out the submodules